src/main/scala/com.snowplowanalytics.maxmind.iplookups/IpLookups.scala

```diff
@@ -65,27 +103,18 @@ object IpLookups {
  *
  * Inspired by:
  * https://github.com/jt6211/hadoop-dns-mining/blob/master/src/main/java/io/covert/dns/geo/IpLookups.java
- *
- * @param geoFile Geographic lookup database file
- * @param ispFile ISP lookup database file
- * @param domainFile Domain lookup database file
- * @param connectionTypeFile Connection type lookup database file
- * @param memCache Whether to use MaxMind's CHMCache
- * @param lruCache Maximum size of SynchronizedLruMap cache
  */
-class IpLookups(
-  geoFile: Option[File] = None,
-  ispFile: Option[File] = None,
-  domainFile: Option[File] = None,
-  connectionTypeFile: Option[File] = None,
-  memCache: Boolean = true,
-  lruCache: Int = 10000
+class IpLookups[F[_]: Sync] private (
+  geoFile: Option[File],
+  ispFile: Option[File],
+  domainFile: Option[File],
+  connectionTypeFile: Option[File],
+  memCache: Boolean,
+  lruCache: Option[LruMap[F, String, IpLookupResult]]
 ) {
 
   // Initialise the cache
-  private val lru =
-    if (lruCache > 0) Some(new SynchronizedLruMap[String, IpLookupResult](lruCache))
-    else None // Of type mutable.Map[String, LookupData]
+  private val lru = lruCache
```

Comment:
?

Reply:
I moved the the call to `LruMap.create` call up to `createFromFiles` and the constructor just takes the Option[LruMap]. I thought it would be better if we didn't need an unsafeRunSync in IpLookups.